File: contest-6084.py
n = int(input())
array = [int(x) for x in input().split()]
div = [1]
for i in range(2, n):
    if n % i == 0:
        div.append(i)

found = False
for d in div:
    error = False
    sum1 = sum(array[0 : d])

    for k in range(1, (n // d)):
        sum2 = sum(array[k * d : (k + 1) * d])

        if sum2 != sum1:
            error = True
            break

    if error is False:
        found = True
        print(n // d)
        break

if found is False:
    print(1)
    

File: contest-6084.cpp
#include <iostream>
#include <vector>
#include <numeric>

using namespace std;

int main()
{
    int n;
    cin >> n;
    vector<long long> array(n);

    for (int i = 0; i < n; ++i)
    {
        cin >> array[i];
    }

    vector<int> div;
    div.push_back(1);

    for (int i = 2; i < n; ++i)
    {
        if (n % i == 0)
        {
            div.push_back(i);
        }
    }

    bool found = false;
    bool error = false;
    long long sum1, sum2;

    for (int d : div)
    {
        error = false;
        sum1 = accumulate(array.begin(), array.begin() + d, 0);

        int t = n / d;
        for (int k = 1; k < t; ++k)
        {
            sum2 = accumulate(array.begin() + (k * d), array.begin() + ((k + 1) * d), 0);
            if (sum2 != sum1)
            {
                error = true;
                break;
            }
        }

        if (error == false)
        {
            found = true;
            cout << t << endl;
            break;
        }
    }

    if (found == false)
    {
        cout << 1 << endl;
    }

    return 0;
}
